silk/tools/wpan_util.py

```diff
@@ -101,3 +101,64 @@ def verify_prefix(node_list, prefix, prefix_len=64, stable=True, priority='med',
         else:
             print "Did not find prefix {} on node {}".format(prefix, node)
             exit(1)
+
+
+def check_parent_on_child_and_childtable_on_parent(parent, children):
+    """Check parent on each child and on parent verify all children are present """
+
+    # Get parent's extended address
+    parent_ext_addr = parent.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+
+    # Verify parent on children
+    for i, child in enumerate(children):
+        # get the extended address(it's length is always 16) of the parent from child
+        thread_parent = child.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        verify(parent_ext_addr == thread_parent)
+        print("***** parent {} has extended address: {}, child {} selected parent: {} *****".
+              format(parent.name, parent_ext_addr, child.name, thread_parent))
+
+    # verify all children are present in selected parent's childtable
+    childTable = parent.wpanctl("get", "get " + wpan.WPAN_THREAD_CHILD_TABLE, 2)
+    childTable = wpan_table_parser.parse_child_table_result(childTable)
+    verify(len(childTable) == len(children))
+
+    counter = 0
+    for i, child in enumerate(children):
+        ext_addr = child.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+
+        for entry in childTable:
+            if entry.ext_address == ext_addr:
+                verify(int(entry.rloc16, 16) == int(child.get(wpan.WPAN_THREAD_RLOC16), 16))
+                verify(int(entry.timeout) == int(child.getprop(wpan.WPAN_THREAD_CHILD_TIMEOUT)))
+                mode = int(child.get(wpan.WPAN_THREAD_DEVICE_MODE), 0)
+
+                if i < len(children) - 2:
+
+                    verify(entry.is_ftd() == (mode & wpan.THREAD_MODE_FLAG_FULL_THREAD_DEV != 0))
```

Comment:
Thanks a lot Jennie for this PR.

As as last discussion to have more sed please change the code lines 137 to 141 as below. Thanks again.
                    verify(child.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_SLEEPY_END_DEVICE)
                else:
                    verify(entry.is_ftd() == (mode & wpan.THREAD_MODE_FLAG_FULL_THREAD_DEV != 0))
                    verify(entry.is_rx_on_when_idle() == (mode & wpan.THREAD_MODE_FLAG_RX_ON_WHEN_IDLE != 0))

Reply:
done